internal/server/routes.go

```diff
@@ -6,9 +6,14 @@ import (
 
 func (s *Server) registerRoutes() *mux.Router {
 	r := mux.NewRouter()
+
+	r.Handle("/authenticate", s.authenticateHandler()).Methods("POST")
+	r.Handle("/users", s.authMiddleware(s.createUserHandler(), true, false)).Methods("POST")
```

Comment:
Why is authentication optional? Someone could just spam new users all over the place. I think you should have to be verified to create new users.

Reply:
So people can register themselves and then somebody can verify them.

Reply:
Oh duh, that makes sense.